service/src/main/java/uk/gov/hmcts/reform/fpl/service/cafcass/api/CafcassApiDocumentService.java

```diff
@@ -10,8 +10,9 @@
 @RequiredArgsConstructor(onConstructor = @__(@Autowired))
 public class CafcassApiDocumentService {
     private final SecureDocStoreService secureDocStoreService;
+    private final CafcassSystemUserService cafcassSystemUserService;
 
     public byte[] downloadDocumentByDocumentId(String documentId) throws IllegalArgumentException, EmptyFileException {
-        return secureDocStoreService.downloadDocument(documentId);
+        return secureDocStoreService.downloadDocument(documentId, cafcassSystemUserService.getUserToken());
```

Comment:
Can we not use the calling user's token here like we normally do when calling the docstore (on requestData)? That should be the cafcass system user, saves logging in again/if ever an accidental bug allows the non-docstore user to login.

Reply:
The IDAM /userinfo endpoint will only return user name but not the other details (e.g. user roles) if the token is generated via password grant. 
CCD / CDAM will throw exception if they can't grab the user role with the tokens.

That's why we need to get the token via login, such that IDAM will return a full userinfo